Problem2.py

```diff
@@ -0,0 +1,38 @@
+# Time Complexity :O(n^m
+# Space Complexity :O(n)
+#  Did this code successfully run on Leetcode :yes
+# Any problem you faced while coding this :no
```

Comment:
Good Job! Think on the space complexity. There is a small error. Let us know if you need further explanation.

Reply:
This will be still O(n) right because I am using a map. And approximately it will take O(n) space where n is the number of diagonals.